service/history/replication/raw_task_converter.go

```diff
@@ -222,6 +222,12 @@ func convertSyncHSMReplicationTask(
 			// HSM can be updated after workflow is completed
 			// so no check on workflow state here.
 
+			if mutableState.HasBufferedEvents() {
```

Comment:
is it possible to check this when generating the sync state replication task?

Reply:
That doesn't really matter because even when there's no buffered events when generating the replication task, there could be buffered events when converting. Similarly even when there's buffered event when generating the task, could might be no buffered events when converting.